tests/integration_test.go

```diff
@@ -116,67 +114,34 @@ func testExampleFile(t *testing.T, cfg generator.Config, fileName string) {
 			goldenFileName := filepath.Join(filepath.Dir(fileName), outputName)
 			t.Logf("Using golden data in %s", mustAbs(goldenFileName))
 
-			goldenData, err := ioutil.ReadFile(goldenFileName)
+			goldenData, err := os.ReadFile(goldenFileName)
 			if err != nil {
 				if !os.IsNotExist(err) {
 					t.Fatal(err)
 				}
 				goldenData = source
 				t.Log("File does not exist; creating it")
-				if err = ioutil.WriteFile(goldenFileName, goldenData, 0655); err != nil {
+				if err = os.WriteFile(goldenFileName, goldenData, 0655); err != nil {
 					t.Fatal(err)
 				}
 			}
-			if diff, ok := diffStrings(t, string(goldenData), string(source)); !ok {
-				t.Fatal(fmt.Sprintf("Contents different (left is expected, right is actual):\n%s", *diff))
-			}
+			require.Equal(t, string(goldenData), string(source))
 		}
 	})
 }
 
 func testFailingExampleFile(t *testing.T, cfg generator.Config, fileName string) {
 	t.Run(titleFromFileName(fileName), func(t *testing.T) {
-		generator, err := generator.New(cfg)
+		g, err := generator.New(cfg)
 		if err != nil {
 			t.Fatal(err)
 		}
-		if err := generator.DoFile(fileName); err == nil {
+		if err := g.DoFile(fileName); err == nil {
 			t.Fatal("Expected test to fail")
 		}
 	})
 }
 
-func diffStrings(t *testing.T, expected, actual string) (*string, bool) {
```

Comment:
ack this may or may not be desired. Personally require.Equal has been helpful enough, but I could understand if you want to keep a more diff utility.